sky/task.py

```diff
@@ -44,7 +45,7 @@
     'a list of node ip addresses (List[str]). Got {run_sig}')
 
 
-def _is_valid_name(name: str) -> bool:
+def _is_valid_name(name: Optional[str]) -> bool:
```

Comment:
A little bit curious why it could pass mypy check previously... Maybe we should file an issue to add type notation for these task-related functions? A lof of function still doesn't have a notation, like `estimate_runtime` or `get_resources`.

Reply:
Yes, we should. Mypy will ignore the functions/variables without type annotation, so probably skip some callers with name that is None.